Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzToolsFramework/API/EntityCompositionNotificationBus.h>
+#include <AzToolsFramework/API/ViewportEditorModeTrackerNotificationBus.h>
+#include <AzToolsFramework/ComponentMode/EditorComponentModeBus.h>
+#include <AzToolsFramework/Entity/EntityTypes.h>
+#include <AzToolsFramework/ViewportUi/Button.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiRequestBus.h>
+
+namespace AZ
+{
+    class Component;
+}
+namespace AzToolsFramework
+{
+    namespace ComponentModeFramework
+    {
+
+        enum class AddOrRemoveComponent : AZ::u8
```

Comment:
nit: It's fine to remove `: AZ::u8` here, it's not necessary and we don't really care about the size in this case